src/datachain/lib/dc.py

```diff
@@ -1027,7 +1027,11 @@ def to_pandas(self, flatten=False) -> "pd.DataFrame":
 
         transposed_result = list(map(list, zip(*self.results())))
         data = {tuple(n): val for n, val in zip(headers, transposed_result)}
-        return pd.DataFrame(data)
+        df = pd.DataFrame(data)
+        # Sort the dataframe to avoid a PerformanceWarning about unsorted indexing.
+        df.sort_index(axis=0, inplace=True, sort_remaining=True)
+        df.sort_index(axis=1, inplace=True, sort_remaining=True)
+        return df
```

Comment:
I moved this sorting to only run for the test where this warning is generated. Having unsorted dataframes can lead to strange performance problems, such as O(N) instead of O(1) operations, so this may end up being a problem for users in the future.